### PR TITLE
feat(camelot): add support for swaps on camelot V3

### PR DESCRIPTION
**Pull request by mmackz**
This PR introduces support for camelots V3 swap router into the `camelot` plugin.

V3 swap router contract - https://arbiscan.io/address/0x1f721e2e82f6676fce4ea07a5958cf098d339e18

This also fixes a bug with the current implementation where some token to ETH transactions were not being counted.

### CODE REVIEW

File: packages/camelot/src/Camelot.ts [hunk 1]
```diff
@@ -6,105 +6,147 @@ import {
...
+      ],
+    },
     value: ethUsedIn ? amountIn : undefined,
     input: {
-      $abi: [...CAMELOT_ABI, ...PARASWAP_ABI],
       $or: [
         {
           // camelotV2 swap
-          to: recipient,
-          path: buildPathQuery(ethUsedIn ? Tokens.WETH : tokenIn, tokenOut),
```
Comment (mmackz) [inline packages/camelot/src/Camelot.ts:54]: This was an issue with the current plugin implementation. if tokenOut is selected as ETH, it needs to be converted to WETH in the path array.

Comment (Quazia) [reply packages/camelot/src/Camelot.ts:54]: Thanks for the callout

File: packages/camelot/src/Camelot.ts [hunk 2]
```diff
@@ -6,105 +6,147 @@ import {
...
+          $abi: PARASWAP_ABI,
+          $or: [
+            {
+              // simpleswap, directUniV3Swap, directCurveSwap
+              data: {
+                fromToken: tokenInOrEth,
+                fromAmount: amountIn,
+                toAmount: amountOut,
+                toToken: tokenOutOrEth,
+                partner: '0x353D2d14Bb674892910685520Ac040f560CcBC06',
```
Comment (Quazia) [inline packages/camelot/src/Camelot.ts:105]: Let's use a const for this

**Review by Quazia — APPROVED** [review]
LGTM

**Comment by Quazia** [comment]
LGTM

**Pull request merged**